src/lib/action/action.c

```diff
@@ -123,25 +128,37 @@ int singularity_action_do(int argc, char **argv) {
     }
     free(target_pwd);
 
-    if ( action == ACTION_SHELL ) {
-        singularity_message(DEBUG, "Running action: shell\n");
-        action_shell_do(argc, argv);
-    } else if ( action == ACTION_EXEC ) {
-        singularity_message(DEBUG, "Running action: exec\n");
-        action_exec_do(argc, argv);
-    } else if ( action == ACTION_RUN ) {
-        singularity_message(DEBUG, "Running action: run\n");
-        action_run_do(argc, argv);
-    } else if ( action == ACTION_TEST ) {
-        singularity_message(DEBUG, "Running action: test\n");
-        action_test_do(argc, argv);
-    } else if ( action == ACTION_START ) {
-        singularity_message(DEBUG, "Running action: start\n");
-        action_start_do(argc, argv);
-    } else if ( action == ACTION_STOP ) {
-        singularity_message(DEBUG, "Running action: stop\n");
-        action_stop_do(argc, argv);
+    child = singularity_fork();
```

Comment:
@bauerm97 and I spoke about this on Slack. This is for the post process rebuild of the environment metadata. I asked Michael to refactor this so it 1. doesn't require a fork, 2. doesn't update the metadata after every command (even if --writable was set). I would prefer either non-state and glob at runtime or only do this after bootstrap/import.